pkg/helpers/api.go

```diff
@@ -0,0 +1,101 @@
+package helpers
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capzV1alpha3 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capiV1alpha3 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/pkg/label"
+)
+
+const (
+	credentialDefaultName = "credential-default"
+	credentialNamespace   = "giantswarm"
+)
+
+// GetClusterFromMetadata returns the Cluster object (if present) using the object metadata.
+func GetAzureClusterFromMetadata(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*capzV1alpha3.AzureCluster, error) {
+	// Check if "cluster.x-k8s.io/cluster-name" label is set.
+	if obj.Labels[capiV1alpha3.ClusterLabelName] == "" {
+		err := microerror.Maskf(invalidObjectError, "Label %q must not be empty for object %q", capiV1alpha3.ClusterLabelName, obj.GetSelfLink())
+		return nil, err
+	}
+
+	return GetAzureClusterByName(ctx, c, obj.Namespace, obj.Labels[capiV1alpha3.ClusterLabelName])
+}
+
+// GetAzureClusterByName finds and return a AzureCluster object using the specified params.
+func GetAzureClusterByName(ctx context.Context, c client.Client, namespace, name string) (*capzV1alpha3.AzureCluster, error) {
+	azureCluster := &capzV1alpha3.AzureCluster{}
+	key := client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}
+
+	if err := c.Get(ctx, key, azureCluster); err != nil {
+		return nil, err
+	}
+
+	return azureCluster, nil
+}
+
+func GetCredentialSecretFromMetadata(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*v1alpha1.CredentialSecret, error) {
+	// r.Logger.LogCtx(ctx, "level", "debug", "message", "finding credential secret")
+
+	organization, exists := obj.GetLabels()[label.Organization]
+	if !exists {
+		return nil, microerror.Mask(missingOrganizationLabel)
+	}
+
+	secretList := &corev1.SecretList{}
+	{
+		err := c.List(
+			ctx,
+			secretList,
+			client.InNamespace(credentialNamespace),
+			client.MatchingLabels{
+				label.App:          "credentiald",
+				label.Organization: organization,
+			},
+		)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	// We currently only support one credential secret per organization.
+	// If there are more than one, return an error.
+	if len(secretList.Items) > 1 {
+		return nil, microerror.Mask(tooManyCredentialsError)
+	}
+
+	// If one credential secret is found, we use that.
+	if len(secretList.Items) == 1 {
+		secret := secretList.Items[0]
+
+		credentialSecret := &v1alpha1.CredentialSecret{
+			Namespace: secret.Namespace,
+			Name:      secret.Name,
+		}
+
+		// r.Logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found credential secret %s/%s", credentialSecret.Namespace, credentialSecret.Name))
```

Comment:
logging is commented out

Reply:
Oh yes,  I forgot to pass in logger here. Will fix.

Reply:
Added logging.